### PR TITLE
Scroll native minting

### PR DESCRIPTION
**Pull request by jtfirek**
- Rebasing the Scroll native minting changes onto `jtdev/feat/sync-pool-files` branch

### CODE REVIEW

File: contracts/NativeMinting/L2SyncPoolContracts/L2ScrollSyncPoolETHUpgradeable.sol [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    function initialize(
+        address l2ExchangeRateProvider,
+        address rateLimiter,
+        address tokenOut,
+        uint32 dstEid,
+        address messenger,
+        address receiver,
+        address delegate
+    ) external virtual initializer {
+
```
Comment (jtfirek) [inline contracts/NativeMinting/L2SyncPoolContracts/L2ScrollSyncPoolETHUpgradeable.sol:55]: I have already deployed the the scroll sync pool with this additional line. Would you like me to remove line and redeploy? The deployments are easy. All I have to do is run `DeployConfigureL1.s.sol` and `DeployConfigureL2.s.sol`

**Comment by jtfirek** [comment]
added audit!

File: contracts/NativeMinting/L2SyncPoolContracts/L2ScrollSyncPoolETHUpgradeable.sol (hunk 1 above)
Comment (seongyun-ko) [reply contracts/NativeMinting/L2SyncPoolContracts/L2ScrollSyncPoolETHUpgradeable.sol:55]: NP!

**Comment by seongyun-ko** [comment]
can u fix the conflicts?

**Pull request merged**